test/sql/upsert/upsert_lambda.test

```diff
@@ -0,0 +1,52 @@
+# name: test/sql/upsert/upsert_lambda.test
+# group: [upsert]
+
+statement ok
+pragma enable_verification;
+
+statement ok
+create or replace table foo(bar int primary key, baz varchar, other varchar[], other2 int);
+
+statement ok
+insert into foo
+select 1, '', ['a', 'b', 'c'], 7
+on conflict do
+update set
+  baz = list_reduce(
+    excluded.other,
+    (x, y) -> x || ' || ' || y
+  )
+;
+
+statement ok
+insert into foo
+select 1, '', ['a', 'b', 'c'], 3
+on conflict do
+update set
+  baz = list_reduce(
+    excluded.other,
+    (x, y) -> x || ' || ' || y
+  )
+;
+
+query IIII
+select * from foo;
+----
+1	a || b || c	[a, b, c]	7
+
+statement ok
+insert into foo
+select 1, '', ['1', '2'], 9
+on conflict do
+update set
+  baz = list_reduce(
+    excluded.other,
+    (x, y) -> x || ' || ' || y
```

Comment:
Skipping qualification for all expressions in a lambda function is likely not sufficient. Lambda functions can also refer to columns not used in the lambda parameters, e.g. `x || baz`. Could we add a test for this?

Reply:
Yea you're right, then it becomes ambiguous
which isn't the worst thing, that it needs to be explicitly qualified when used in the lambda

Nothing really breaks because of this, the only this is that some user experience is lost
In other places when `baz` is used, it gets qualified to `foo.baz` implicitly to avoid the ambiguity

```sql
statement error
insert into foo
select 1, '', ['a', 'b', 'c'], 7
on conflict do
update set
  baz = list_reduce(
    excluded.other,
    (x, y) -> x || baz || y
  )
;
----
Ambiguous reference to column name "baz" (use: "excluded.baz" or "foo.baz")
```

This is the behavior now

Reply:
That sounds fine to me yeah